### PR TITLE
Single manager bug

### PR DESCRIPTION
**Pull request by mwiktorin**
I added a class WorkerAgent for all agents which will be managed by a manager. We should think about the DoughRester here.
ManagerAgents now find their workers and we have managers for each bakery.
See #38

### CODE REVIEW

File: src/main/java/org/awesomebakery/agents/workers/OvenAgent.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+	}
+
+	@Override
+	public String getServiceType() {
+		return SERVICE_TYPE;
+	}
+
+	@Override
+	public void onMessageReceived(ACLMessage message) {
+		// TODO Auto-generated method stub
```
Comment (sonarcube) [inline src/main/java/org/awesomebakery/agents/workers/OvenAgent.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/org/awesomebakery/agents/managers/DoughPrepManager.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	}
+
+	@Override
+	public String getServiceType() {
+		return SERVICE_TYPE;
+	}
+	
+	@Override
+	public void onMessageReceived(ACLMessage message) {
+		// TODO Auto-generated method stub
```
Comment (sonarcube) [inline src/main/java/org/awesomebakery/agents/managers/DoughPrepManager.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/org/awesomebakery/agents/workers/KneadingAgent.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+	}
+
+	@Override
+	public String getServiceType() {
+		return SERVICE_TYPE;
+	}
+
+	@Override
+	public void onMessageReceived(ACLMessage message) {
+		// TODO Auto-generated method stub
```
Comment (sonarcube) [inline src/main/java/org/awesomebakery/agents/workers/KneadingAgent.java]: INFO Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

File: src/main/java/org/awesomebakery/agents/managers/ManagerAgent.java [hunk 4]
```diff
@@ -1,17 +1,62 @@
...
+
+	private static List<String> getGuidsFromEntityList(List<? extends Entity> list) {
+		List<String> result = new ArrayList<>();
+		for (Entity entity : list) {
+			result.add(entity.getGuid());
+		}
+		return result;
+	}
+
+	public abstract List<? extends Entity> getWorkerList();
```
Comment (sonarcube) [inline src/main/java/org/awesomebakery/agents/managers/ManagerAgent.java]: CRITICAL Remove usage of generic wildcard type. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1452)

**Comment by sonarcube** [comment]
SonarQube analysis reported 6 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [DoughPreparerAgent.java#L25](https://github.com/HBRS-MAAS/project-awesomebakery/blob/ed3e31165aab97209185c462f3d830dc0629222e/src/main/java/org/awesomebakery/agents/workers/DoughPreparerAgent.java#L25): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TruckAgent.java#L24](https://github.com/HBRS-MAAS/project-awesomebakery/blob/ed3e31165aab97209185c462f3d830dc0629222e/src/main/java/org/awesomebakery/agents/workers/TruckAgent.java#L24): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by sonarcube** [comment]
SonarQube analysis reported 5 issues
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [DoughPreparerAgent.java#L25](https://github.com/HBRS-MAAS/project-awesomebakery/blob/8d05cd59ea8ce1a559e206983e1006358492d94a/src/main/java/org/awesomebakery/agents/workers/DoughPreparerAgent.java#L25): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TruckAgent.java#L24](https://github.com/HBRS-MAAS/project-awesomebakery/blob/8d05cd59ea8ce1a559e206983e1006358492d94a/src/main/java/org/awesomebakery/agents/workers/TruckAgent.java#L24): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Comment by moriarty** [comment]
@mwiktorin what was the critical error?

Sometimes you can justify ignoring them...

**Comment by mwiktorin** [comment]
@moriarty 
The issue was that there was a wildcard in the return type. Sonarqube said wildcards in return types of functions are not good, they should only be used in parameters

**Comment by sonarcube** [comment]
SonarQube analysis reported 5 issues
* ![INFO][INFO] 5 info

Watch the comments in this conversation to review them.

#### 2 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![INFO][INFO] [DoughPreparerAgent.java#L25](https://github.com/HBRS-MAAS/project-awesomebakery/blob/3dd2ff8e73eb7417248b37001425a68c0e17c2ac/src/main/java/org/awesomebakery/agents/workers/DoughPreparerAgent.java#L25): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)
1. ![INFO][INFO] [TruckAgent.java#L24](https://github.com/HBRS-MAAS/project-awesomebakery/blob/3dd2ff8e73eb7417248b37001425a68c0e17c2ac/src/main/java/org/awesomebakery/agents/workers/TruckAgent.java#L24): Complete the task associated to this TODO comment. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1135)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'

**Pull request merged**